wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -345,8 +345,10 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
         Future.failed(new RuntimeException(
           s"Attempting to spend an ImmatureCoinbase ${out.outPoint.hex}, this should not be possible until it is confirmed."))
       case TxoState.ConfirmedSpent =>
-        Future.failed(new RuntimeException(
-          s"Attempted to mark an already spent utxo ${out.outPoint.hex} with a new spending tx ${spendingTxId.hex}"))
+        if (!out.spendingTxIdOpt.contains(spendingTxId)) {
```

Comment:
Hmm, did this happen to you in normal node operations? Why is this needed to be added here?

Reply:
This happened when I moved back the sync height without deleting my db. Imo this should probably be kept incase we reprocess a tx that is confirmed.